server/routes/bitbucket/bitbucket_selection.py

```
@@ -196,6 +228,264 @@ def clear_workspace_selection(user_id):
+    webhook_url: str, webhook_events, secret: str,
+) -> bool:
+    """Best-effort API creation of the repo webhook; returns True on success.
+
+    Needs repo admin (and webhook scopes the connector token usually lacks) —
+    manual setup is the primary path, so failure here is informational only.
+    An existing hook with the same URL counts as success (its uuid is
+    recorded too, so disable/disconnect can still delete it).
+    """
+    def _store_hook_uuid(hook_uuid: str) -> None:
+        with db_pool.get_admin_connection() as conn:
+            with conn.cursor() as cur:
```

Ruff (0.16.1)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)